File: Data Structures & Algorithms/minimize-the-maximum-difference-of-pairs/submission-1.cpp
class Solution {
public:
    int minimizeMax(vector<int>& nums, int p) {
        sort(nums.begin(),nums.end());
        int l=0,r=nums[nums.size()-1]-nums[0],m;

        while(r>=l){
            m=(l+r)/2;
            int t=0;
            for(int i=1;i<nums.size();i++){
                if(nums[i]-nums[i-1]<=m){
                    i++;
                    t++;
                }
            }
            if(t>=p)
            r=m-1;
            else l=m+1;
        }
        return l;
      
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-1.java
class Solution {
    int nums[];
    int target;
    List<List<Integer>> soln = new ArrayList<>();
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        this.nums=nums;
        this.target=target;
        List<Integer> list = new ArrayList<>();
        backTrack(0,0,list);
        return soln;
    }

    void backTrack(int i,int sum,List<Integer> list){
        if(sum==target){
            soln.add(new ArrayList<>(list));
            return;
        }
        if(i<nums.length && sum<target){
            int temp = nums[i];
            list.add(temp);
            backTrack(i,sum+temp,list);
            list.remove(list.size()-1);
            backTrack(i+1,sum,list);
        }
    }
}
